src/utils.c

```diff
@@ -263,3 +263,21 @@ gchar * r_realpath(const gchar *path)
 
 	return g_strdup(rpath);
 }
+
+gboolean r_whitespace_removed(gchar *str)
+{
+	gsize len;
+
+	if (str == NULL)
+		return FALSE;
+
+	len = strlen(str);
+
+	if (len == 0)
+		return FALSE;
+
+	g_strstrip(str);
+
+	/* If the last char is now \0, some whitespace was removed. */
+	return str[len-1] == '\0';
```

Comment:
`g_strstrip()` is implemented as a composition of `g_strchug()` and `g_strchomp()` and does not give any guarantees with respect to how the shortened string looks like compared to the original one. The test is not correct in general: looking at [`g_strchug()`'s current implementation](https://github.com/GNOME/glib/blob/mainline/glib/gstrfuncs.c#L2256) shows that applying `r_whitespace_removed()` to a string with more than one leading space is going to give the wrong result.

Reply:
You're right. Premature optimization...